sdk/cosmosdb/cosmos/src/ClientContext.ts

```diff
@@ -757,12 +757,13 @@ export class ClientContext {
       options: requestContext.options,
       partitionKeyRangeId: requestContext.partitionKeyRangeId,
       useMultipleWriteLocations: this.connectionPolicy.useMultipleWriteLocations,
-      partitionKey: requestContext.partitionKey,
+      partitionKey: requestContext.partitionKey !== undefined ? convertToInternalPartitionKey(requestContext.partitionKey) : undefined //TODO: Move this check from here to PartitionKey
```

Comment:
you can also do `requestContext.partitionKey ?? convertToInternalPartitionKey(requestContext.partitionKey)`

Reply:
The desired output of this expression is 
- `undefiend` -> if value is `undefined`
- convertToInternalPartitionKey(partitionKey) if not.

Actually a `a = b && func(b)` equivalent was needed, but since partitionKey could be `boolean false` also can't use it.

With `Nullish Coalescing` it would evaluate to `partitionKey` instead of convertToInternalPartitionKey(partitionKey). I know it looks a bit verbose but couldn't find any concise expression of this use case.